main.tf

```diff
@@ -0,0 +1,45 @@
+/**
+ * # AWS EKS External DNS Terraform module
+ *
+ * A terraform module to deploy the [ExternalDNS](https://kubernetes-sigs.github.io/external-dns/latest) on Amazon EKS cluster.
+ *
+ * [![Terraform validate](https://github.com/lablabs/terraform-aws-eks-external-dns/actions/workflows/validate.yaml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-external-dns/actions/workflows/validate.yaml)
+ * [![pre-commit](https://github.com/lablabs/terraform-aws-eks-external-dns/workflows/pre-commit.yml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-external-dns/actions/workflows/pre-commit.yml)
+ *
+ */
+
+locals {
+  addon = {
+    name      = "external-dns"
+    namespace = "kube-system"
+
+    helm_chart_version = "1.15.1"
+    helm_repo_url      = "https://kubernetes-sigs.github.io/external-dns"
+  }
+
+  addon_irsa = {
+    (local.addon.name) = {
+      irsa_policy         = var.irsa_policy != null ? var.irsa_policy : one(data.aws_iam_policy_document.this[*].json)
+      irsa_policy_enabled = var.irsa_policy_enabled != null ? var.irsa_policy_enabled : true # Its false by default in module and we need to generate custom policy
+    }
+  }
+
+  addon_values = yamlencode({
```

Comment:
Review values according to the Helm chart https://artifacthub.io/packages/helm/external-dns/external-dns?modal=values-schema. In this PR we changed Helm chart from Bitnami to SIG one with a different set of values.